gcache/src/gcache_rb_store.cpp

```diff
@@ -337,11 +341,20 @@ namespace gcache
         assert(size_used_ >= bh->size);
         size_used_ -= bh->size;
 
+        // free just allocated, never used buffer (move from size_used_ -> size_free_)
         if (SEQNO_NONE == bh->seqno_g)
         {
-            empty_buffer(bh);
+            // move it as it was really released prior to discard.
+            size_rnd_ += bh->size;
             discard (bh);
+            empty_buffer(bh);
         }
+        else
+        {
+            // free (release) used buffer. move size_used_ -> size_rnd_
+            size_rnd_ += bh->size;
```

Comment:
This happens in both cases, and as the first statement. You could do it before the if, then we won't ever need the else clause.

Reply:
fixed